### PR TITLE
Rev C++ header for greater consistency

### PR DESCRIPTION
**Pull request by PeterConstable**
Revised some field names for greater consistency internally and with the OT/OFF spec names.

Also revised description of ClipList to use ArrayOf as elsewhere in the C++ header.

### CODE REVIEW

File: colr-gradients-spec.md [hunk 1]
```diff
@@ -561,8 +560,7 @@ typedef ArrayOf<Offset32<Paint>, uint32> LayerList;
 struct ClipList
 {
   uint8                 format;  // Set to 1.
-  uint32                numClips;
-  Clip                  clips[/*numClips*/];  // Clip records, sorted by startGlyphID
+  ArrayOf<Clip>         clips;  // Clip records, sorted by startGlyphID
```
Comment (behdad) [inline colr-gradients-spec.md:563]: `ArrayOf` uses a 16bit count; we had 32 here. Call it `Array32Of`.

@rsheeter do we need more than 16bit though?

Comment (anthrotype) [reply colr-gradients-spec.md:563]: previously we had `ArrayOf<Clip, uint32> ClipList`
https://github.com/googlefonts/colr-gradients-spec/pull/347/files#diff-458fde56ab00bcf68739126026367f5098e0ea122f477fbd2ec03d870eba40e7L561

> do we need more than 16bit though?

probably not, but why restrict oneself, if one really wishes to assign one individual clip box to every single base colr glyph and there are > 65k of them.. It's only the count type (16bit vs 32bit), so +2 bytes in total per ClipList table (and there's only one per font)

Comment (PeterConstable) [reply colr-gradients-spec.md:563]: Sorry, I had overlooked that ArrayOf uses uint16.

For a count that occurs once, I don't see any reason _not_ to use uint32. I'll restore that part.

**Comment by behdad** [comment]
Agree that keeping 32bit is good.

Feel free to also makes sizes explicit everywhere in this code. In HarfBuzz we now have `Offset16To/Offset24To/Offset32To/Array16Of/Array32Of`.

File: colr-gradients-spec.md (hunk 1 above)
Comment (PeterConstable) [reply colr-gradients-spec.md]: I looked again and realized ArrayOf could be used with uint32 as a second arg, so changing to that.

**Comment by PeterConstable** [comment]
If others way to change ArrayOf with optional count type to Array16Of / Array32Of, etc. I have no opinion and will let others do that.

**Review by rsheeter — APPROVED** [review]

**Pull request merged**